src/explorer/appSettingsNodes.ts

```diff
@@ -0,0 +1,206 @@
+            return 'Key must have at least one non-whitespace character.'; 
+        }
+
+        if (this._settings.properties && this._settings.properties[newKey] && newKey !== oldKey) {
+            return `Setting "${newKey}" already exists.`; 
+        }
+    }
+
+    protected get WebSiteManagementClient(): WebSiteManagementClient {
+        return this._websiteClient;
+    }
+
+    private get azureAccount(): AzureAccountWrapper {
+        return this.getTreeDataProvider<AppServiceDataProvider>().azureAccount;
+    }
+
+    private getSettings(): Promise<WebSiteModels.StringDictionary> {
+        const webApps = this.WebSiteManagementClient.webApps;
+        return this._isSlot ? webApps.listApplicationSettingsSlot(this.site.resourceGroup, this._siteName, this._slotName) : 
+            webApps.listApplicationSettings(this.site.resourceGroup, this._siteName);
+    }
+
+    private applySettings(): Promise<WebSiteModels.StringDictionary> {
+        const webApps = this.WebSiteManagementClient.webApps;
+        const updateTask = this._isSlot ? webApps.updateApplicationSettingsSlot(this.site.resourceGroup, this._siteName, this._settings, this._slotName) :
+            webApps.updateApplicationSettings(this.site.resourceGroup, this._siteName, this._settings);
+        return updateTask;
+    }
+
+    private generateTreeItems(settings: WebSiteModels.StringDictionary): AppSettingNode[] {
+        const children: AppSettingNode[] = [];
+
+        if (settings.properties) {
+            for (let key in settings.properties) {
+                children.push(new AppSettingNode(
+                    key,
+                    settings.properties[key],
+                    this.getTreeDataProvider(),
+                    this
+                ));
+            }
+        }
+
+        return children.sort((a, b) => a.label.localeCompare(b.label));
+    }
+}
+
+export class AppSettingNode extends NodeBase {
+    constructor(private key: string,
+        private value: string,
+        treeDataProvider: AppServiceDataProvider,
+        parentNode: NodeBase) {
+        super(`${key} : ${value}`, treeDataProvider, parentNode);
+    }
+
+    getTreeItem(): TreeItem {
+        return {
+            label: this.label,
+            collapsibleState: TreeItemCollapsibleState.None,
+            contextValue: "applicationSettingItem",
+            iconPath: { 
+                light: path.join(__filename, '..', '..', '..', '..', 'resources', 'light', 'Item_16x_vscode.svg'),
+                dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'dark', 'Item_16x_vscode.svg')
+            }
+        }
+    }
+
+    async edit(): Promise<void> {
+        const oldKey = this.key;
+        const newKey = await vscode.window.showInputBox({
+            ignoreFocusOut: true,
+            prompt: 'Enter setting key',
+            value: this.key,
+            validateInput: v => (<AppSettingsNode>this.getParentNode()).validateNewKeyInput(v, oldKey)
```

Comment:
I feel like all of these should either be `this.getParentNode<AppSettingsNode>()` or `<AppSettingsNode>this.parentNode`, right?